lightblue/src/main/java/org/esbtools/eventhandler/lightblue/client/BulkLightblueRequester.java

```diff
@@ -373,14 +375,36 @@ public U get() throws InterruptedException, ExecutionException {
             return result;
         }
 
-        // TODO(ahenning): This ignores the timeout because we aren't doing work in another thread
-        // I'm not sure that anyone will ever care, but if they did, we would do the same lazy
-        // stuff, but just do it in another thread which we could interrupt if it took too long.
-        // In that case we may have to build in interrupt checking during request processing.
         @Override
-        public U get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException,
-                TimeoutException {
-            return get();
+        public U get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+            if (cancelled) {
+                throw new CancellationException();
+            }
+
+            if (!completed) {
+                ExecutorService executor = Executors.newSingleThreadExecutor();
+                try {
+                    Future<?> submittedTask = executor.submit(() -> {
+                        completer.triggerFutureCompletion();
+                    });
```

Comment:
Up to you, but you can do a method reference instead of a lambda here if you wanted

```
... = executor.submit(completere::triggerFutureCompletion);
```